### DISCUSSION TITLE
Writable commit buffers / hunks

### DISCUSSION THREAD

**Discussion by nicolas-graves** [magit/magit#5014] [Ideas]
I really like wgrep for quickly editing a variable in multiple files at once. 
I often have to modify a commit to rework it before pushing anything. They are easy to see in a buffer with magit.
I would love to be able to edit it directly in a writable commit buffer, instead of hitting "modify" and go precisely to the place action happened.
Adding a key "modify-in-place" or something along those lines would be a tremendous benediction in my use of magit.  

In the meantime, setq an option to jump to the place of the commit when hitting "modify" would be beneficial too.

**Comment by tarsius** [comment]
> Adding a key "modify-in-place" or something along those lines would be a tremendous benediction in my use of magit.

That won't be easy but it is on my long lists of things that I might eventually implement.

> In the meantime, setq an option to jump to the place of the commit when hitting "modify" would be beneficial too.

(I assume that by `hitting "modify"`, you mean `C-c C-e` (`magit-diff-edit-hunk-commit`).)

I thought that was already happening, but apparently not. That should be a bit easier to implement because we already do jump to the correct location in other situations, so we can probably just adapt some existing code. (We should always do it, not just as opt-in behavior.)

Note to self: investigate if there are other places where that should be done. Can we generalize this so that point is always adjusted in all file-visiting buffers when another commit is checked out? That, however, would be a larger undertaking, especially if we also address issues such as point ending up inside invisible areas (e.g., because of `outline-minor-mode`).

**Comment by nicolas-graves** [comment]
On 2023-09-28 05:32, Jonas Bernoulli wrote:

>> Adding a key "modify-in-place" or something along those lines would be a tremendous benediction in my use of magit.
>
> That won't be easy but it is on my long lists of things that I might eventually implement.

Thanks for the quick answer.
I figured that, hence the second part ;)

>
> (I assume that by `hitting "modify"`, you mean `C-c C-e` (`magit-diff-edit-hunk-commit`).)
>

I actually meant choosing "modify" in the rebase transient menu, I
didn't really used `magit-diff-edit-hunk-commit`. Will give a try, it
seems to jump to the file but not to the line of the hunk.

-- 
Best regards,
Nicolas Graves

**Reply by tarsius** [reply]
Another workflow that you might not be familiar with is that you can simply make the change on top of HEAD and then assign it to some other commit either automatically using `c x` (after staging), or manually using `c F` (if the former doesn't work or you would rather not have to double check).

Of course that (as well as `magit-diff-edit-hunk-commit` and the feature that you are proposing) can result in a conflict. Actually if there is a conflict, there will be two conflicts, going in "opposite directions", which makes it harder to avoid making mistakes when deciding, which changes should remain in any given commit that is being rewritten during rebase.

For that reason `r m` is the safest option.

**Comment by tarsius** [comment]
I do not plan to support editing diffs in place. It would be an interesting feature and not completely off the table, but it's unlikely I'll implement it. Alternative workflows are available, some of which I have mentioned above, and while sometimes editing the diff directly would be a good or even the best approach, it is a can of worms.
